feat: add how-to guide for using LLMs and MCP in app plugins

sd2k opened the pull request:
This commit adds a new how-to guide for app plugin authors,
demonstrating how to use the `@grafana/llm` package to make
requests to LLMs, and how to make use of the Grafana MCP
server embedded into the LLM app.

I've combined these into one guide since the concepts are closely
related but I could split them out if that's preferred.

sd2k committed:
feat: add how-to guide for using LLMs and MCP in app plugins

This commit adds a new how-to guide for app plugin authors,
demonstrating how to use the `@grafana/llm` package to make
requests to LLMs, and how to make use of the Grafana MCP
server embedded into the LLM app.

I've combined these into one guide since the concepts are closely
related but I could split them out if that's preferred.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/grafana/plugin-tools?pullRequest=1864) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/grafana/plugin-tools?pullRequest=1864) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/grafana/plugin-tools?pullRequest=1864) it.</sub>

sd2k commented:
Good call, it does require a fairly recent version of the LLM app for the MCP stuff so I'll take a look at that. Needs slightly updating for some changes we've recently made too. Thanks for reminding me I created this :sweat_smile:

sd2k committed:
Include info on versions; improve MCP examples

academo reviewed (commented):
Do we have any available documentation for these llm packages? maybe a reference or more complete documentation of its functionality?

It'll be nice to add links to the documentation of the functions in the places you are using them in the examples.

sd2k committed:
Replace throws with early returns

academo commented:
@sd2k this looks correct but I have the feeling the wording doesn't go according to the way we write things in the docs.

My advice:

* feed the writers toolkit https://grafana.com/docs/writers-toolkit/ and other examples from our plugin docs to an LLM and this text and ask for a review
* ask a human writer to help you review this

the content itself is good to me

sd2k committed:
Update LLM/MCP how-to guide to be more procedural

The doc now uses step-by-step instructions to guide users along
rather than jumping straight into code examples, and separates the
topics a bit better.

academo commented on docusaurus/docs/how-to-guides/app-plugins/use-llms-and-mcp.md:
```diff
@@ -0,0 +1,644 @@
+Before you begin, ensure you have:
+
+- The [Grafana LLM app] plugin installed and enabled (version 0.22 or later for MCP features)
+- A Grafana app plugin development environment set up
```
add a link to our app tutorial?

sd2k committed:
Include a link to the 'build an app plugin' tutorial
